### ISSUE TITLE
Incorrectly Named Parameter Not Caught in Azure Portal Compilation/Logs

### ISSUE THREAD

**Issue by MisterJames** [Azure/azure-functions-host#2400]
#### Investigative information

- Timestamp: 2018-02-08T17:42:58.2887108Z;  (This is the latest entry in the blog logs for a PUT operation)
- Function App version (1.0 or 2.0-beta): Runtime version: 1.0.11510.0 (~1)
- Function App name: not relevant 
- Function names : ResizeProductImage
- Invocation ID: (none)
- Region: West US

#### Repro steps

1. Create function in csharp (DLL project) with BlobTrigger attribute
2. Add a `string` parameter called `imageName`
3. Deploy to Azure
4. Upload image to blob container

#### Expected behavior

After I complete step 3 I should see an invocation of the function. There is a runtime failure because the imageName can't be bound.

I should see an error in the portal indicating the issue.

Locally, the Azure CLI tools indicate a parameter problem.

#### Actual behavior

No invocation of the function occurs.

## Other Notes

`AzureStorageAccount` is the connection used. It is correctly configured in the portal complete with blob location and connection string. This was copied from elsewhere in the portal. The corresponding blob containers exist in the storage account.

Other functions in this app are working correctly off of their respective triggers (such as queue or timer).

Logs are enabled and the `$logs` folder exists in the blob container of my storage account. I see my blob operations listed (and have a timestamp above from a working write to the blob log).

The function is never invoked. The first line of the method is a logging statement which never appears in the logs.

Here is a snapshot of the container:

![image](https://user-images.githubusercontent.com/1197383/35990439-9e657f26-0cc9-11e8-94d6-83cc43a1f405.png)

<details>
<summary>Source</summary>

```csharp
public static class ResizeProductImage
    {
        [FunctionName("ResizeProductImage")]
        public static void Run(
            [BlobTrigger("product-images-full/{name}", Connection = "AzureStorageAccount")]Stream inputImage, 
            [Blob("product-images-thumbs/{name}", FileAccess.Write, Connection = "AzureStorageAccount")]Stream outputImage,
            string imageName, 
            TraceWriter log)
        {
            log.Info($"C# Blob trigger function Processed blob\n Name:{imageName} \n Size: {inputImage.Length} Bytes");
            var settings = new ImageResizer.ResizeSettings
            {
                MaxWidth = 500,
                Format = "png"
            };

            ImageResizer.ImageBuilder.Current.Build(inputImage, outputImage, settings);

        }
    }
```
</details>

**Comment by MisterJames** [comment]
I think this can be tracked as a different issue...I'm not getting any errors in the portal, but I am locally for missing parameter names.

I had renamed the `name` parameter to `imageName` just before publishing (the function was working locally). It didn't occur to me off the hop that this would be set as `name` in the env vars (I assume?) when the function executes.

Locally, I just ran it again and got a runtime error, pointing me to the `imageName` parameter. I reverted the change and the binding works again.
